internal/usecase/repo/order.go

```diff
@@ -0,0 +1,123 @@
+package repo
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/jackc/pgx/v5"
+	"gophermart/internal/entity"
+	"gophermart/pkg/logger"
+	"gophermart/pkg/postgres"
+	"time"
+)
+
+type OrderRepo struct {
+	*postgres.Postgres
+}
+
+func NewOrder(pg *postgres.Postgres) *OrderRepo {
+	return &OrderRepo{pg}
+}
+
+func (orr *OrderRepo) Get(ctx context.Context, number entity.OrderNumber) (*entity.Order, error) {
+	query := `select user_id, number, status, created_at from user_order where number = $1`
+	o := &entity.Order{}
+	err := orr.Pool.QueryRow(ctx, query, number).Scan(&o.UserID, &o.Number, &o.Status, &o.UploadedAt)
+	if err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return nil, entity.ErrOrderNotFound
+		}
+		return nil, fmt.Errorf("OrderRepo - Get - orr.Pool.QueryRow: %w", err)
+	}
+	return o, nil
+}
+
+func (orr *OrderRepo) Add(ctx context.Context, order entity.Order) (*entity.Order, error) {
```

Comment:
```suggestion
func (orr *OrderRepo) Add(ctx context.Context, order entity.Order) error {
```
зачем возвращать данные, которые только что передали в аргументе? Если изменяем данные - нам интересно прошло ли это успешно

если хочется получить данные - нужно сделать метод например `order := GetByID(orderID)`